src/python/TaskWorker/MasterWorker.py

```diff
@@ -524,16 +560,29 @@ def algorithm(self):
         self.restartQueuedTasks()
         self.logger.debug("Master Worker Starting Main Cycle.")
         while not self.STOP:
-            selection_limit = self.config.TaskWorker.task_scheduling_limit
-            if not self._selectWork(limit=selection_limit):
-                self.logger.warning("No tasks selected.")
-            else:
-                self.logger.info("Work selected successfully.")
+            is_canary = self.config.TaskWorker.is_canary
+            canary_name = self.config.TaskWorker.canary_name
```

Comment:
are we protected against this key to be missing in config file ? See your comment at line 181

Reply:
so comment in line 181 is wrong ?

Reply:
Oh you mean 581. It is technically true because you will not "specify" tw_canary_name in the canary TW config file...however indocker.pp will set it as "...CHANGE ME..." when it is not specified. That is the default value. and since it doesn't start with crab...only master will call the runCanary function.

Reply:
I will change the comment to clarify

Reply:
I tried to follow the same route as other hiera variables. Having a default value helps us skip putting to many if-else statements in the code.

Reply:
Yes, i can put it under the else of is_canary check.